dot15d4/src/phy/driver.rs

```diff
@@ -0,0 +1,123 @@
+use core::task::Context;
+use core::task::Poll;
+use std::future::Future;
+
+#[cfg_attr(feature = "std", derive(Debug))]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[derive(PartialEq, Clone, Copy)]
+pub enum Error {
+    CCAFailed,
+    ACKFailed,
+    InvalidStructure,
+    RadioError,
+}
+
+/// Should be given as an argument to the task that will run the network protocol.
+/// This trait allows to abstract over channels in async executors.
+pub trait Driver {
+    /// Waits until there is something to be transmitted
+    fn transmit(&self) -> impl Future<Output = PacketBuffer>;
+    /// Hold until the buffer is received successfully
+    fn received(&self, buffer: PacketBuffer) -> impl Future<Output = ()>;
+    /// Hold until the buffer is received successfully
+    fn error(&self, error: Error) -> impl Future<Output = ()>;
+}
+
+/// A buffer that is used to store 1 frame.
+#[cfg_attr(feature = "std", derive(Debug))]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[derive(Clone, PartialEq)]
+pub struct PacketBuffer {
+    /// The data of the frame that should be transmitted over the radio
+    pub buffer: [u8; 128],
```

Comment:
Could you add a note to explain why it is 128 bytes instead of 127? A full IEEE 802.15.4 frame is 127 bytes. I guess it is 128 bytes for the nRF52840 radio?

Reply:
Yes this is the reason. Should I increase this to higher number (like 130) to support other radios. This would then be enough for radios that would need to explicitly store a length and the crc.

Reply:
The 127 bytes already contain the CRC (FCS).